enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,649 @@
+workloads they will be able to do that via the existing Performance Profile.
+Resizing partition size will not cause any issues after installation.
+
+In order to implement this enhancement we are proposing changing 3 components
+defined below.
+
+1. Openshift API - ([Infrastructure
+   Status](https://github.com/openshift/api/blob/81fadf1ca0981f800029bc5e2fe2dc7f47ce698b/config/v1/types_infrastructure.go#L51))
+
+   - The change in this component is to store a global identifier if we have
+     partitioning turned on.
+
+2. Admission Controller ([management cpus
+   override](https://github.com/openshift/kubernetes/blob/a9d6306a701d8fa89a548aa7e132603f6cd89275/openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/doc.go))
+   in openshift/kubernetes.
+   - This change will be in support of checking the global identifier in order
+     to modify the pod spec with the correct `requests`.
+3. The
+   [Performance Profile Controller](https://github.com/openshift/cluster-node-tuning-operator/blob/master/docs/performanceprofile/performance_profile.md)
+   part of [Cluster Node Tuning
+   Operator](https://github.com/openshift/cluster-node-tuning-operator)
+   - This change will support adding the ability to explicitly pin
+     `Infrastructure` workloads.
+   - This change will support updating a global identifier when workload
+     partitioning is detected on the nodes.
+
+### Openshift API - Infrastructure Status
+
+We will need to maintain a status flag to be able to identify if a cluster we
+are operating in has been setup for partitioning or not. Since this identifier
+signifies that a cluster's infrastructure is setup for workload partitioning we
+feel that this information should be part of the `Infrastructure Status`
+resource. This identifier will be an enum that will be set during installation
+and during an upgrade for existing single-node deployments we will set it via
+the Node Tuning Operator.
+
+We propose that in the current implementation we only support either `None` or
+`AllNodes`.
+
+```go
+type InfrastructureStatus struct {
+	...
+  // cpuPartitioning expresses if the cluster nodes have CPU Set partitioning turned on.
+  // The default of None means that no CPU Set partitioning is set.
+  // If AllNodes is set that indicates that all the nodes have partitioning set on
+  // and workloads might be pinned to specific CPU Sets depending on the configurations
+  // set via the Node Tuning Operator and the Performance Profile API
+	// +kubebuilder:default=None
+	// +kubebuilder:validation:Enum=None;AllNodes
+	CPUPartitioning PartitioningMode `json:"cpuPartitioning"`
+}
+
+// PartitioningMode defines the CPU partitioning mode of the nodes.
+type CPUPartitioningMode string
+
+const (
+  // No partitioning is applied.
+	CPUPartitioningNone        CPUPartitioningMode = "None"
+  // All nodes are configured for partitioning.
+	CPUPartitioningAllNodes    CPUPartitioningMode = "AllNodes"
+)
+```
+
+### Openshift Installer
```

Comment:
We talked about trying to do this without any changes to the installer by having the user edit the manifests it creates. Did we abandon that approach, or has it not made it into this draft?

Reply:
Sorry about that Doug, that is removed as we discussed, local file save snafu on my end.